packages/flutter/lib/src/services/text_input.dart

```diff
@@ -627,7 +627,7 @@ abstract class TextInputClient {
 /// See also:
 ///
 ///  * [TextInput.attach]
-class TextInputConnection {
+class TextInputConnection with ChangeNotifier {
```

Comment:
I tend to think this might be a bit too all-encompassing and doesn't leave us room to expand later (if we're implying that the only reason why this object would call you back is to disconnect). I'd just add an explicit listener.

Reply:
Doesn't `ChangeNotifier` mean "something has changed, query me to determine what you need to do"?